Kernel/Net/TCPSocket.h

```diff
@@ -126,6 +126,9 @@ class TCPSocket final : public IPv4Socket {
     static RefPtr<TCPSocket> from_endpoints(const IPv4Address& local_address, u16 local_port, const IPv4Address& peer_address, u16 peer_port);
 
     RefPtr<TCPSocket> create_client(const IPv4Address& local_address, u16 local_port, const IPv4Address& peer_address, u16 peer_port);
+    void set_originator(RefPtr<TCPSocket> originator) { m_originator = originator; }
```

Comment:
It's not okay to call `set_originator(nullptr)`, so let's not use a `RefPtr` or `TCPSocket*`
I also don't think we're ever passing ownership of the `TCPSocket` from the caller to this function, so this should be...
```cpp
void set_originator(TCPSocket& originator) { m_originator = originator; }
```

In general terms:
`RefPtr` or `NonnullRefPtr` for arguments where the caller is passing ownership to the callee, `T*` or `T&` otherwise.
`T*` for arguments where `nullptr` is a valid value, `T&` if not.